crates/katana/rpc/rpc/tests/starknet.rs

```diff
@@ -172,5 +176,171 @@ async fn test_send_declare_and_deploy_legacy_contract() {
         class_hash
     );
 
-    sequencer.stop().expect("failed to stop sequencer");
+    drop(katana_runner)
+}
+
+sol!(
+    #[allow(missing_docs)]
+    #[sol(rpc)]
+    StarknetContract,
+    "tests/test_data/solidity/StarknetMessagingLocalCompiled.json"
+);
+
+sol!(
+    #[allow(missing_docs)]
+    #[sol(rpc)]
+    Contract1,
+    "tests/test_data/solidity/Contract1Compiled.json"
+);
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_messaging_l1_l2() {
+    // Prepare Anvil + Messaging Contracts
+    let anvil_runner = AnvilRunner::new().await.unwrap();
+
+    let contract_strk = StarknetContract::deploy(anvil_runner.provider()).await.unwrap();
+    let strk_address = contract_strk.address();
+
+    assert_eq!(
+        contract_strk.address(),
+        &Address::from_str("0x5fbdb2315678afecb367f032d93f642f64180aa3").unwrap()
+    );
+
+    let contract_c1 = Contract1::deploy(anvil_runner.provider(), *strk_address).await.unwrap();
+
+    assert_eq!(
+        contract_c1.address(),
+        &Address::from_str("0xe7f1725e7734ce288f8367e1bb143e90bb3f0512").unwrap()
+    );
+
+    // Prepare Katana + Messaging Contract
+    let messagin_config = json!({
+        "chain": "ethereum",
+        "rpc_url": anvil_runner.endpoint,
+        "contract_address": contract_strk.address().to_string(),
+        "sender_address": "0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266",
+        "private_key": "0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80",
+        "interval": 2,
+        "from_block": 0
+    });
+    let serialized_json =
+        serde_json::to_string_pretty(&messagin_config).expect("Failed to serialize JSON");
+
+    // Write JSON string to a file
+    let mut file =
+        File::create("tests/test_data/anvil-messaging.json").expect("Failed to create file");
+    file.write_all(serialized_json.as_bytes()).expect("Failed to write to file");
+
+    let katana_runner =
+        KatanaRunner::new_with_messaging("tests/test_data/anvil-messaging.json".to_string())
+            .unwrap();
+    let starknet_account = katana_runner.account(0);
+
+    let path: PathBuf = PathBuf::from("tests/test_data/cairo_l1_msg_contract.json");
+    let (contract, compiled_class_hash) =
+        common::prepare_contract_declaration_params(&path).unwrap();
+
+    let class_hash = contract.class_hash();
+    let res =
+        starknet_account.declare(Arc::new(contract), compiled_class_hash).send().await.unwrap();
+
+    let receipt = TransactionWaiter::new(res.transaction_hash, starknet_account.provider())
+        .with_tx_status(TransactionFinalityStatus::AcceptedOnL2)
+        .await
+        .expect("Invalid tx receipt");
+
+    assert_eq!(receipt.finality_status(), &TransactionFinalityStatus::AcceptedOnL2);
+
+    assert!(
+        starknet_account
+            .provider()
+            .get_class(BlockId::Tag(BlockTag::Latest), class_hash)
+            .await
+            .is_ok()
+    );
```

Comment:
these asserts are to check that the tx is successful and the contract is deployed, right?. lets add a comment about that for better readability.